src/consumer.cpp

```diff
@@ -48,6 +52,23 @@ void Consumer::rebalance_proxy(rd_kafka_t*, rd_kafka_resp_err_t error,
     static_cast<Consumer*>(opaque)->handle_rebalance(error, list);
 }
 
+TopicPartitionList Consumer::get_matching_partitions(TopicPartitionList&& partitions,
+                                                     const vector<string>& topics) {
+    TopicPartitionList matches;
+    for (const auto& topic : topics) {
+        for (auto& partition : partitions) {
+            bool match = equal(topic.begin(), topic.end(), partition.get_topic().begin(),
```

Comment:
Won't this blow up if `topic.size() > partition.get_topic().size()`?

Reply:
Yeah looks like the signature was improved in C++17 with a fourth end iterator. I'll put a boundary check.